Modules/Core/Common/src/itkMetaDataDictionary.cxx

```diff
@@ -21,30 +21,46 @@ namespace itk
 {
 MetaDataDictionary
 ::MetaDataDictionary()
+  : m_Dictionary( std::make_shared<MetaDataDictionaryMapType>() )
 {
-  m_Dictionary = new MetaDataDictionaryMapType;
 }
 
 MetaDataDictionary
 ::~MetaDataDictionary()
 {
-  delete m_Dictionary;
-  m_Dictionary = nullptr;
 }
 
 MetaDataDictionary
 ::MetaDataDictionary(const MetaDataDictionary & old)
+  // perform shallow copy, so m_Dictionary is shared
+  :  m_Dictionary( old.m_Dictionary )
 {
-  m_Dictionary = new MetaDataDictionaryMapType;
-  *m_Dictionary = *( old.m_Dictionary );
 }
 
+MetaDataDictionary
+::MetaDataDictionary( MetaDataDictionary && rr)
+  :  m_Dictionary( std::move(rr.m_Dictionary) )
+{
```

Comment:
Still looks like explicitly defaulted move-constructor (`= default`) would just do the right thing, OK?